### DISCUSSION TITLE
Downsize image without rough edges on text.

ImageMagick 7.1.1-6 Q16-HDRI x86_64
Debian GNU/Linux

### DISCUSSION THREAD

**Discussion by replica9000** [ImageMagick/ImageMagick#6260] [Help]
I'm updating my music collection to use 1000x1000 cover art images.  The images come from different sources, varying from 1200x1200 and as high as 3000x3000, and almost all will contain text.

Example Image:
[3000x3000](https://f4.bcbits.com/img/a1893181602_0.jpg)
[2000x2000](https://m.media-amazon.com/images/I/81q9D156TNL.jpg)

I'm using this to resize the images.
`convert-7 'Sky Void Of Stars.jpg' -resize 1000 -filter triangle -adaptive-sharpen 0.5x0.5 -define jpeg:dct-method=float -interlace none -quality 90 -strip 'Sky Void Of Stars-1.jpg'`

Even before using sharpen, the text tends to have the rough staircase effect. 
[Output Image](https://replica9000.bizland.com/album-art/Sky%20Void%20Of%20Stars.jpg)

Is there a way to improve the text in downsized images?

**Comment by snibgo** [comment]
> Even before using sharpen, the text tends to have the rough staircase effect.

I can't see any "rough staircase effect".

> Is there a way to improve the text in downsized images?

I can't see any problem. For downsizing plus sharpening, I suggest the method shown in [Resampling with halo minimization](http://im.snibgo.com/resamphm.htm).

I also suggest not starting from images that have had JPEG compression.

**Reply by replica9000** [reply]
I see it more in the angled parts of the letters.

[Original](https://replica9000.bizland.com/album-art/110609-1.png)
[Resized](https://replica9000.bizland.com/album-art/110613-1.png)

**Comment by snibgo** [comment]
What command(s) did you use? Perhaps you shrunk the image then enlarged it. Why did you do that?

IM has many methods for resizing. If you shrink and enlarge, you will lose detail. That will always happen, but the detail that is lost depends on the methods used. IM has `-magnify` which may suit your needs. See https://imagemagick.org/script/command-line-options.php#magnify

**Reply by replica9000** [reply]
I'm not enlarging the images, only shrinking them.  The target size is 1000px, but the sources can be anything larger as there's no one source that has all the images I need.

What I'm asking is if there might be a better method to downsize album artwork images that include text (as opposed to outdoor/scenic images), or if the command (see first post) should be adjusted according to the size of the source image.

If it matters, my display is 4k, so maybe the difference is more noticable to me?

Thank you.

**Comment by snibgo** [comment]
> I'm not enlarging the images, only shrinking them.

You linked to two images, "Original" and "Resized". "Original" is 304x78. "Resized" is 305x79, so it is slightly larger. It appears to have been downsized then upsized. But perhaps you did something different. If you don't say what you did, I can't comment.

**Reply by replica9000** [reply]
Those were just cropped out of the images from the first post comparing the text from the [original image](https://f4.bcbits.com/img/a1893181602_0.jpg) and the [resized image](https://replica9000.bizland.com/album-art/Sky%20Void%20Of%20Stars.jpg).
